cmake/OpenCVDetectCUDA.cmake

```diff
@@ -86,12 +86,17 @@ if(CUDA_FOUND)
     unset(CUDA_ARCH_PTX CACHE)
   endif()
 
-  SET(DETECT_ARCHS_COMMAND "${CUDA_NVCC_EXECUTABLE}" ${CUDA_NVCC_FLAGS} "${OpenCV_SOURCE_DIR}/cmake/checks/OpenCVDetectCudaArch.cu" "--run")
-  if(WIN32 AND CMAKE_LINKER) #Workaround for VS cl.exe not being in the env. path
-    get_filename_component(host_compiler_bindir ${CMAKE_LINKER} DIRECTORY)
-    SET(DETECT_ARCHS_COMMAND ${DETECT_ARCHS_COMMAND} "-ccbin" "${host_compiler_bindir}")
+  if(CUDA_HOST_COMPILER)
+    LIST(APPEND CUDA_NVCC_FLAGS -ccbin ${CUDA_HOST_COMPILER})
```

Comment:
`CUDA_HOST_COMPILER` is path to executable, but not directory as it's done in else branch.

Reply:
I wanted to avoid specifying -ccbin twice.

Reply:
> `CUDA_HOST_COMPILER` is path to executable, but not directory as it's done in else branch.

If the user specifies CUDA_HOST_COMPILER, we wouldn't need to help him detect cl.exe

> `CUDA_HOST_COMPILER` is path to executable, but not directory as it's done in else branch.

4.2.1.11. --compiler-bindir directory (-ccbin)

Specify the directory in which the compiler executable resides.

**The host compiler executable name can be also specified to ensure that the correct host compiler is selected**. In addition, driver prefix options (--input-drive-prefix, --dependency-drive-prefix, or --drive-prefix) may need to be specified, if nvcc is executed in a Cygwin shell or a MinGW shell on Windows.